src/transformers/models/blenderbot/modeling_flax_blenderbot.py

```diff
@@ -782,12 +791,21 @@ def __call__(
         last_hidden_states = outputs[0]
         last_hidden_states = self.layer_norm(last_hidden_states)
 
+        # update the last element in `hidden_states` after applying `layernorm` above
+        hidden_states = None
+        if output_hidden_states:
+            hidden_states = outputs[1]
+            hidden_states = hidden_states[:-1] + (last_hidden_states,)
+
         if not return_dict:
-            return (last_hidden_states,) + outputs[1:]
+            if not output_hidden_states:
+                return (last_hidden_states,) + outputs[1:]
+            else:
+                return (last_hidden_states, hidden_states) + outputs[2:]
```

Comment:
same comment as above

Reply:
Hi, is this better

https://github.com/huggingface/transformers/blob/415eb3c0ff9cfd68f3ba25122370324220f01b87/src/transformers/models/blenderbot/modeling_flax_blenderbot.py#L720-L722

(when it returns tuple + when it needs extra processing as in this PR, I always have trouble to make it cleaner. Things get much easier if the internal components return dict or named tuple, and only change the format at the top level components for the users - but I don't think we are going to do so, at least not soon)

Reply:
This looks good!

> Things get much easier if the internal components return dict or named tuple, and only change the format at the top level components for the users

This is a good idea, I think we can change the internal modules to only return either dict or Tuple

Reply:
Thanks for the feedback :-) @patil-suraj . I will apply the same change to other places to finish this PR.
About changing  internal components (in general), let's have a discussion later with other members.